unleash/Unleash.swift

```diff
@@ -35,12 +35,31 @@ public class Unleash {
         self.refreshInterval = refreshInterval
         self.strategies = strategies
         
+        register()
+    }
+    
+    private func register() {
         let body: ClientRegistration = ClientRegistration(appName: appName, strategies: strategies)
-        registerService.register(url: URL(string: self.url)!, body: body)
-            .done {_ in
-            }.catch {error in
-                print("error \(error)")
+        attempt(maximumRetryCount: 3, delayBeforeRetry: .seconds(60)) {
```

Comment:
This won't block the user and is expected to basically happen in the "background". We thought about using GCD and adding an exponential backoff but we looked at the other Unleash client SDKs and the majority of them just make the API call once and don't retry. So we decided this will be good for now.

Reply:
Should there be any kind of unit testing around any of this? Or is it too much Apple/framework stuff to worry about whether 1 failure still makes it happen?

Reply:
I was planning on trying to add unit test coverage tonight and see how easy it is.